### ISSUE TITLE
Conv1d ops don't get quantized

### ISSUE THREAD

**Issue by joe-antognini** [tensorflow/tensorflow#22361]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: OS X
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/A
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: 1.10.1
- **Python version**: 2.7.10
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: See below.

### Describe the problem

Currently `tf.contrib.quantize.create_eval_graph` and `tf.contrib.quantize.create_training_graph` will create a quantized version of the graph if the graph contains 2d convolutions.  The pattern matching that these functions use breaks in the case of 1d convolutions, however.  This is because 1d convolutions are transformed to 2d convolutions under the hood with an `ExpandDims` Op, and this Op breaks the expected pattern of Ops used by `_FindLayersToQuantize`.

It would be great if the pattern in `_FindLayersToQuantize` could be changed from

```
weight|folded_weight --> conv|fc --> [batch_to_space_nd] --> ...
```

to

```
weight|folded_weight --> [expand_dims] --> conv|fc --> [batch_to_space_nd] --> ...
```

This should allow conv1d ops to quantize exactly like conv2d ops.

### Source code / logs

```
# Create and quantize a graph with 2d convolution.
X = tf.placeholder(tf.float32, [None, 28, 28, 1])
conv = tf.layers.conv2d(X, 64, 3, padding='same', activation=tf.nn.relu)
tf.contrib.quantize.create_eval_graph()
len([n.name for n in tf.get_default_graph().as_graph_def().node if 'quant' in n.name]) > 0
```

This should be `True`.

```
# Create and quantize a graph with 1d convolution.
X = tf.placeholder(tf.float32, [None, 28, 1])
conv = tf.layers.conv1d(X, 64, 3, padding='same', activation=tf.nn.relu)
tf.contrib.quantize.create_eval_graph()
len([n.name for n in tf.get_default_graph().as_graph_def().node if 'quant' in n.name]) > 0
```

This should be `False`.

**Comment by wt-huang** [comment]
@joe-antognini good point. Just ran the above code snippets and both returned True. For image quantization conv2d is typically used, e.g. in your example. conv1d essentially does the same thing here, unless you want to do text quantization, which would require different methodology.

**Comment by joe-antognini** [comment]
Interesting, what version of Tensorflow did you run those snippets on?  I tried again and I still get `True` from the first snippet and `False` from the second.  The reason I'm interested in quantizing 1d convolution is for audio applications.

**Comment by wt-huang** [comment]
1.10.0

**Comment by joe-antognini** [comment]
@wt-huang, just as a sanity check, did you reset the graph between running the two snippets?

**Comment by wt-huang** [comment]
@joe-antognini I didn't reset the graph between two snippets.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @wt-huang: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by wt-huang** [comment]
Will close this issue soon, chime in if this problem persists.
